### DISCUSSION TITLE
Is this how useQueries should work ?

### DISCUSSION THREAD

**Discussion by craig-harbottle** [TanStack/query#1954] [Q&A]
So, here's what I see is happening : 

1. User clicks on a button to load some measurements. This is the log of useQueries results (and query key): 
<img width="505" alt="Screenshot 2021-03-12 at 03 52 33" src="https://user-images.githubusercontent.com/16166272/110929095-81cf3f80-831f-11eb-84e2-e67328ffd5fd.png">

All good. Data is returned for each individual query and stored against the query key. The three loading states change to success. 

2. User clicks to another measurement. 

<img width="501" alt="Screenshot 2021-03-12 at 04 02 58" src="https://user-images.githubusercontent.com/16166272/110929477-e2f71300-831f-11eb-9778-a156de1825fc.png">

The initial response is 'success' and the data of the previous query is returned.  But thereafter, the response for the measurement data the user clicked the btn for is returned. 

Is this expected behaviour? If so, is there any way to prevent it? I want to be able to show a loader and to do that I need to be able to distinguish between 'success' and 'loading'. But that is complicated when the 'next' request returns some previous 'success' value.

My assumption would be that at step 2. (fetching new data), we should see a loading state and for the data to be reset to undefined (basically 1. again) . Thanks in advance :-)

**Comment by TkDodo** [comment]
how are you "loading measurements" when the user clicks a button? I think you would get your expected behaviour if you change the query-key whenever the user clicks the button, which will create a new cache entry, so it starts with undefined data and loading state again.

if you re-use the same query key (maybe by calling `refetch()`, I don't know...?), you will get the cached data and no `isLoading` state (though `isFetching` will be true). This is the desired behaviour because when refetching the same data (background refetch), it is usually not wanted to unmount the data and show a spinner instead. This is what stale-while-revalidate is all about :)

**Reply by craig-harbottle** [reply]
Hi @TkDodo - thanks for replying :-) 

`how are you "loading measurements" when the user clicks a button?` - list item in an `accordion` 

`I think you would get your expected behaviour if you change the query-key whenever the user clicks the button, which will create a new cache entry, so it starts with undefined data and loading state again.` - me too 🤔 

Here're a at typical query key array. The `measurementName` changes on each btn click : 

<img width="438" alt="Screenshot 2021-03-12 at 12 10 37" src="https://user-images.githubusercontent.com/16166272/110938653-18a1f900-832c-11eb-9e05-0f7c858d77e8.png">

 But that's not the correct data for that measurement. It's from the prev query.

- `refetch()` isn't called, afaik

**Reply by TkDodo** [reply]
can you create a codesandbox reproduction please? I can take a better look at a running example :)

**Reply by craig-harbottle** [reply]
Hi @TkDodo - my issue was fixed with a package update. The issue was prob [this](https://github.com/tannerlinsley/react-query/commit/8a613165fc7dbc5213f61ad75f29ba54d248a0df#diff-1486bf9371a46f2a5a849e437c96eb09808ae49a4eebbd32f312b18bc87caf99R44) one. Thanks for taking the time to reply.
